jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditorDialog.kt

```diff
@@ -0,0 +1,151 @@
+    override fun doValidate(): ValidationInfo? = validator.validateEditorEntries(view)
+    override fun getOKAction(): Action = action
+    override fun doCancelAction() {
+        super.doCancelAction()
+    }
+
+    override fun doOKAction() {
+        // Do nothing, close logic is handled separately
+    }
+
+    private fun getCurrentTime() = Calendar.getInstance().toInstant()
+
+    private fun getRelativeTime(unitOfTime: ChronoUnit?, relTimeNumber: Long): StartEndDate = StartEndDate(getCurrentTime().minus(relTimeNumber, unitOfTime),
+        getCurrentTime())
+
+    private fun getAbsoluteTime(startDate: Date, endDate: Date): StartEndDate = StartEndDate(startDate.toInstant(), endDate.toInstant())
+
+    private fun getFilterQuery(searchTerm: String): String {
+        if (searchTerm.contains("/")) {
+            val regexTerm = searchTerm.replace("/", "\\/")
+            return "fields @message, @timestamp | filter @message like /$regexTerm/"
+        }
+        return "fields @message, @timestamp | filter @message like /$searchTerm/"
+    }
+
+    private fun beginQuerying() {
+        if (!okAction.isEnabled) {
+            return
+        }
+        val funDetails = getFunctionDetails()
+        val queryStartEndDate: StartEndDate
+        queryStartEndDate = (if (funDetails.absoluteTimeSelected) {
+            getAbsoluteTime(funDetails.startDateAbsolute, funDetails.endDateAbsolute)
+        } else {
+            getRelativeTime(relativeTimeUnit[funDetails.relativeTimeUnit], funDetails.relativeTimeNumber.toLong())
+        })
+        val query = if (funDetails.enterQuery) {
+            funDetails.query } else {
+            getFilterQuery(funDetails.searchTerm)
+        }
+        close(OK_EXIT_CODE)
+        queryingLogGroupApiCall.executeStartQuery(queryStartEndDate, funDetails.logGroupName, query)
+    }
+
+    private fun getFunctionDetails(): QueryDetails = QueryDetails(
+        logGroupName = logGroupNames,
+        absoluteTimeSelected = view.absoluteTimeRadioButton.isSelected,
+        startDateAbsolute = view.startDate.date,
+        endDateAbsolute = view.endDate.date,
+        relativeTimeSelected = view.relativeTimeRadioButton.isSelected,
+        relativeTimeUnit = view.relativeTimeUnit.selectedItem.toString(),
+        relativeTimeNumber = view.relativeTimeNumber.text,
+        searchTermSelected = view.searchTerm.isSelected,
+        searchTerm = view.querySearchTerm.text,
+        enterQuery = view.queryLogGroupsRadioButton.isSelected,
+        query = view.queryBox.text
+    )
+
+    private inner class QueryLogGroupOkAction : OkAction() {
+        init {
+            putValue(Action.NAME, message("cloudwatch.logs.query.form.ok_Button"))
+        }
+        override fun doAction(e: ActionEvent?) {
+            super.doAction(e)
+            if (doValidateAll().isNotEmpty()) return
+            beginQuerying()
+        }
+    }
+}
+
+object QueryEditorValidator {
```

Comment:
This function can go into the dialog, we don't gain anything by putting it in a different object

Reply:
The object is referenced later in the tests

Reply:
Tests can call `doValidate` instead?

Reply:
Moved the validateEditorEntries function to QueryEditorDialog class